api/android/api/src/main/java/org/nnsuite/nnstreamer/Tensor.java

```diff
@@ -71,118 +89,204 @@ public void addTensorInfo(int type, @NonNull int[] dimension) {
      *
      * @throws IndexOutOfBoundsException if the given index is invalid
      */
-    public int getTensorType(int index) {
+    public int getType(int index) {
         checkIndexBounds(index);
         return mInfoList.get(index).getType();
     }
 
     /**
      * Sets the tensor dimension
      *
-     * @param index     The index of the tensor information in the list
+     * @param index     The index of the tensor in the list
      * @param dimension The tensor dimension
      *
      * @throws IndexOutOfBoundsException if the given index is invalid
      * @throws IllegalArgumentException if the given dimension is null or invalid
      */
-    public void setTensorDimension(int index, @NonNull int[] dimension) {
+    public void setDimension(int index, @NonNull int[] dimension) {
         checkIndexBounds(index);
         mInfoList.get(index).setDimension(dimension);
     }
 
     /**
      * Gets the tensor dimension of given index.
      *
-     * @param index The index of the tensor information in the list
+     * @param index The index of the tensor in the list
      *
      * @return The tensor dimension
      *
      * @throws IndexOutOfBoundsException if the given index is invalid
      */
-    public int[] getTensorDimension(int index) {
+    public int[] getDimension(int index) {
         checkIndexBounds(index);
         return mInfoList.get(index).getDimension();
     }
 
     /**
-     * Calculates the byte size of tensor data.
+     * Sets a tensor data.
      *
-     * @param index The index of the tensor information in the list
+     * @param index The index of the tensor in the list
+     * @param data  The instance of <code>ByteBuffer</code>
      *
-     * @return The byte size of tensor
+     * @throws IndexOutOfBoundsException if the given index is invalid
+     * @throws IllegalArgumentException if the data is not a byte buffer or the buffer is invalid
+     */
+    public void setData(int index, @NonNull Object data) {
+        if (data == null || !(data instanceof ByteBuffer)) {
+            throw new IllegalArgumentException("Given data is not a byte buffer");
+        }
+
+        setData(index, (ByteBuffer) data);
+    }
+
+    /**
+     * Sets a tensor data.
+     *
+     * @param index The index of the tensor in the list
+     * @param data  The byte array to be set
      *
      * @throws IndexOutOfBoundsException if the given index is invalid
-     * @throws IllegalStateException if data type or dimension is invalid
+     * @throws IllegalArgumentException if given data is invalid
      */
-    public int getTensorSize(int index) {
+    public void setData(int index, @NonNull byte[] data) {
+        if (data == null) {
+            throw new IllegalArgumentException("Given data is null");
+        }
+
+        ByteBuffer buffer = allocateByteBuffer(data.length);
+        buffer.put(data);
+
+        setData(index, buffer);
+    }
+
+    /**
+     * Sets a tensor data.
+     *
```

Comment:
Can not we use @brief for the setData function?

Reply:
To upload library with javadoc (jcenter), brief tag is not supported.